runtime/src/tokio/runtime.rs

```diff
@@ -416,35 +419,35 @@ impl crate::Spawner for Context {
         // Get metrics
         let (_, metric) = spawn_metrics!(self);
 
-        // Track parent-child relationship when supervision is requested
-        let parent_children = if self.model.is_supervised() {
-            Some(self.children.clone())
-        } else {
-            None
-        };
-
-        // Set up the task
-        let executor = self.executor.clone();
-        let dedicated = self.model.is_dedicated();
-        let blocking = self.model.is_blocking();
-        self.model = Model::default();
-
-        // Give spawned task its own empty children list
-        let children = Arc::new(Mutex::new(Vec::new()));
-        self.children = children.clone();
+        // Track supervision before resetting configuration
+        let parent = Arc::clone(&self.tree);
+        let past = self.execution;
+        self.execution = Execution::default();
+        let (child, aborted) = Tree::child(&parent);
+        if aborted {
+            return Handle::closed(metric);
+        }
+        self.tree = child;
```

Comment:
Just to be explicit, this avoids re-parenting because it assigns the task as a child of the context we create via `spawn`, rather than the old method of assigning it as a child of the context that performs the spawn?

old:
```
parent ctx
    |
    + -- parent task
    |
    + -- child ctx
        |
        + -- child task
```

new:
```
parent ctx
    |
    + -- parent task
        |
        + -- child ctx (spawn)
            |
            + -- child task
```